java/compaction/compaction-job-creation/src/test/java/sleeper/compaction/job/creation/CreateCompactionJobsTest.java

```diff
@@ -38,272 +40,354 @@
-                    .outputFile(job.getOutputFile())
-                    .partitionId("C")
-                    .build());
-            verifySetJobForFilesInStateStore(job.getId(), List.of(fileReference3, fileReference4));
-            verifyJobCreationReported(job);
-        });
-    }
+    @Nested
+    @DisplayName("Compact files using strategy")
+    class CompactFilesByStrategy {
+        private final List<CompactionJob> jobs = new ArrayList<>();
+        private final CreateCompactionJobs jobCreator = CreateCompactionJobs.standard(
+                ObjectFactory.noUserJars(), instanceProperties,
+                new FixedTablePropertiesProvider(tableProperties),
+                new FixedStateStoreProvider(tableProperties, stateStore),
+                jobs::add, jobStatusStore);
 
-    @Test
-    public void shouldCreateCompactionJobAfterPreSplittingFiles() throws Exception {
-        // Given
-        List<Partition> partitions = new PartitionsBuilder(schema)
-                .rootFirst("A")
-                .splitToNewChildren("A", "B", "C", "ddd")
-                .buildList();
-        setPartitions(partitions);
-        FileReferenceFactory fileReferenceFactory = fileReferenceFactory();
-        FileReference fileReference1 = fileReferenceFactory.partitionFile("A", "file1", 200L);
-        FileReference fileReference2 = fileReferenceFactory.partitionFile("A", "file2", 200L);
-        setFileReferences(List.of(fileReference1, fileReference2));
-
-        // When
-        List<CompactionJob> jobs = createJobs();
-
-        // Then
-        assertThat(jobs).satisfiesExactlyInAnyOrder(job -> {
-            assertThat(job).isEqualTo(CompactionJob.builder()
-                    .jobId(job.getId())
-                    .tableId(tableProperties.get(TABLE_ID))
-                    .inputFiles(List.of("file1", "file2"))
-                    .outputFile(job.getOutputFile())
-                    .partitionId("B")
-                    .build());
-            verifySetJobForFilesInStateStore(job.getId(), List.of(
-                    referenceForChildPartition(fileReference1, "B"),
-                    referenceForChildPartition(fileReference2, "B")));
-            verifyJobCreationReported(job);
-        }, job -> {
-            assertThat(job).isEqualTo(CompactionJob.builder()
-                    .jobId(job.getId())
-                    .tableId(tableProperties.get(TABLE_ID))
-                    .inputFiles(List.of("file1", "file2"))
-                    .outputFile(job.getOutputFile())
-                    .partitionId("C")
-                    .build());
-            verifySetJobForFilesInStateStore(job.getId(), List.of(
-                    referenceForChildPartition(fileReference1, "C"),
-                    referenceForChildPartition(fileReference2, "C")
-            ));
-            verifyJobCreationReported(job);
-        });
-    }
+        @BeforeEach
+        void setUp() {
+            jobs.clear();
+        }
```

Comment:
Each test gets a new instance of the test class, so it'll be a fresh list every time. This can be removed.

Reply:
Done, thanks